Index: netifd-2019-08-05-5e02f944/interface.c
===================================================================
--- netifd-2019-08-05-5e02f944.orig/interface.c
+++ netifd-2019-08-05-5e02f944/interface.c
@@ -424,7 +424,11 @@ interface_main_dev_cb(struct device_user
 		interface_set_link_state(iface, false);
 		break;
 	case DEV_EVENT_TOPO_CHANGE:
-		interface_proto_event(iface->proto, PROTO_CMD_RENEW, false);
+	/* This renews the dhcp lease when the bridge adds/deletes a
+	 * new interface. It causes some dhcp servers to fail in
+	 * case where there are many interfaces being added to the
+	 * bridge frequently. Disabling this for now. */
+	/*	interface_proto_event(iface->proto, PROTO_CMD_RENEW, false); */
 		return;
 	default:
 		break;
